### ISSUE TITLE
Hippo with improper nomad credentials returns 500 server error

### ISSUE THREAD

**Issue by bacongobbler** [deislabs/hippo#773]
If I do not point Hippo to a nomad server, this is what I see:

```
$ export HIPPO_PASSWORD=Passw0rd!
$ export HIPPO_USERNAME=administrator
$ export HIPPO_URL=https://localhost:5309
$ export BINDLE_URL=http://localhost:8080/v1
$ hippo register -k
Registered administrator
$ spin deploy
thread 'main' panicked at 'Unable to create Hippo App: ResponseError(ResponseContent { status: 500, content: "{\"type\":\"https://tools.ietf.org/html/rfc7231#section-6.6.1\",\"title\":\"An error occurred while processing your request.\",\"status\":500}", entity: Some(UnknownValue(Object({"type": String("https://tools.ietf.org/html/rfc7231#section-6.6.1"), "title": String("An error occurred while processing your request."), "status": Number(500)}))) })', /home/bacongobbler/code/spin/src/commands/deploy.rs:117:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

No server error logs are provided.

```
><> dotnet run
Building...
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: https://localhost:5309
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /home/bacongobbler/code/hippo/src/Web/
info: Hippo.Application.Accounts.Commands.CreateAccountCommand[0]
      Hippo Request: CreateAccountCommand   Hippo.Application.Accounts.Commands.CreateAccountCommand
info: Hippo.Application.Accounts.Commands.CreateTokenCommand[0]
      Hippo Request: CreateTokenCommand   Hippo.Application.Accounts.Commands.CreateTokenCommand
info: Hippo.Application.Apps.Commands.CreateAppCommand[0]
      Hippo Request: CreateAppCommand d1648c47-c4fe-49f9-99a1-486640baa2aa administrator Hippo.Application.Apps.Commands.CreateAppCommand
info: Hippo.Infrastructure.Services.DomainEventService[0]
      Publishing domain event. Event - CreatedEvent`1
info: Hippo.Application.Apps.EventHandlers.AppCreatedEventHandler[0]
      Hippo Domain Event: CreatedEvent`1
info: Hippo.Application.Channels.Commands.CreateChannelCommand[0]
      Hippo Request: CreateChannelCommand d1648c47-c4fe-49f9-99a1-486640baa2aa administrator Hippo.Application.Channels.Commands.CreateChannelCommand
info: Hippo.Infrastructure.Services.DomainEventService[0]
      Publishing domain event. Event - CreatedEvent`1
info: Hippo.Application.Channels.EventHandlers.ChannelCreatedEventHandler[0]
      Hippo Domain Event: CreatedEvent`1
info: Hippo.Application.Channels.Commands.CreateChannelCommand[0]
      Published event: CreatedEvent`1
info: Hippo.Application.Apps.EventHandlers.AppCreatedEventHandler[0]
      Hippo Domain Event: CreatedEvent`1
info: Hippo.Application.Revisions.Commands.ImportRevisionsCommand[0]
      Hippo Request: ImportRevisionsCommand d1648c47-c4fe-49f9-99a1-486640baa2aa administrator Hippo.Application.Revisions.Commands.ImportRevisionsCommand
info: Hippo.Infrastructure.Services.DomainEventService[0]
      Publishing domain event. Event - CreatedEvent`1
info: Hippo.Application.Revisions.EventHandlers.RevisionCreatedEventHandler[0]
      Hippo Domain Event: CreatedEvent`1
info: Hippo.Application.Revisions.EventHandlers.RevisionCreatedEventHandler[0]
      Channel 882f8d5d-43b8-4c03-b606-37dd772af7cd changed its active revision to 08083069-09fc-4312-98e1-37b2ca767074
info: Hippo.Application.Revisions.EventHandlers.UpdateRevisionDetails[0]
      Hippo Domain Event: CreatedEvent`1
```

Hippo should error out on startup saying it cannot connect to the nomad API. It should also report any connection errors.

**Comment by michelleN** [comment]
I'm getting this 500. Wondering if it might because of the same issue.
```
examples/http-rust [main] $ hippo app add spin-hello-world spin-hello-world
Error: {"type":"https://tools.ietf.org/html/rfc7231#section-6.6.1","title":"An error occurred while processing your request.","status":500}
```

Are there docs you can point me to for how to point hippo to Nomad?

**Comment by bacongobbler** [comment]
At the moment, there is no documentation. The docs are currently out of date. Working on fixing that today with the nomad-local-demo. :)

For the time being, have a look at https://github.com/deislabs/hippo/blob/bdb2396b81da3271c1d66debc1fc6f4aafcc43a6/src/Infrastructure/Services/NomadService.cs#L20-L21 for the two pieces of configuration Hippo needs to communicate with nomad. In short:

- Nomad:Url is the listening address for nomad
- Nomad:Secret is the authentication token used to communicate with the nomad API (I think this can be ignored if you booted nomad with `nomad agent -dev`

**Comment by bacongobbler** [comment]
https://github.com/deislabs/hippo/blob/bdb2396b81da3271c1d66debc1fc6f4aafcc43a6/src/Web/appsettings.json#L13-L19

**Comment by michelleN** [comment]
How come creating an app in Hippo requires ensuring there is a valid connection to Nomad? I'm not sure at a high level I understand what is or should be happening here. Would you mind breaking it down a bit for me? Can I set up a Hippo app without actually deploying anything to Hippo itself? Just have the abstraction exist and all set up before actually pushing a bindle?

**Comment by bacongobbler** [comment]
A couple of recent feature additions to hippo are at play here.

- #708 introduced a "default" channel named "Production" as soon as the application is created.
- #706 forces the user to choose a storage ID that exists in the bindle server.
- #706 also imports bindle IDs straight from the bindle-server, so that an application's Revision list is populated on creation.

Therefore, when you create a new application, Hippo pulls all available tags from bindle-server and deploys the most recent revision to nomad. By extension, you cannot set up a Hippo app without actually deploying anything to Hippo, because at least one bindle ID must be present in bindle-server.

The nice thing about this new workflow is that you do

```
$ spin bindle push ...
pushed helloworld/1.0.0
$ hippo app add helloworld helloworld
```

And bam, your application is up and running. The downside is that this whole workflow requires everything to be up and serving on boot, and we do not have a "mock" nomad service like we did prior to #724.

Earlier this morning, @gogeterobert, @StefanNedelcu and I discussed how it'd be nice to tie bindle-server and nomad into the [healthcheck endpoint](https://github.com/deislabs/hippo/blob/f755e2730939c392305af6a2cd02a10505f35eb5/src/Web/Program.cs#L28-L29). From there, we could inform the user that certain services are unhealthy (i.e. nomad is unreachable). For console users, that could mean we report those issues in the server logs... But I'm not sure if .NET's healthcheck system reports healthcheck issues into the server logs or not.
